Lib/test/test_array.py

```diff
@@ -1665,5 +1665,11 @@ def test_tolist(self, size):
         self.assertEqual(ls[:8], list(example[:8]))
         self.assertEqual(ls[-8:], list(example[-8:]))
 
+    def test_gh_128961(self):
+        a = array.array('i')
+        it = iter(a)
+        list(it)
+        it.__setstate__(0)
```

Comment:
I wonder if we should also verify that `it` still raises `StopIteration` after this call. What do you think?

Reply:
It does, test added.